crates/zksync/inspectors/src/trace.rs

```diff
@@ -271,6 +267,13 @@ impl InspectorExt for TraceCollector {
                 }
             }
 
+            // We increment the depth after the start of the span to avoid being deeper for single
+            // calls like when using cast call.
```

Comment:
This feels more like a comment for the PR, explaining the _change_, not what we are doing here

Reply:
```suggestion
            // We increment the depth for inner calls as normally traces are processed
            // during execution, where the environment takes care of updating the context
```

Reply:
I took the liberty of explaining here because the function is complicated enough already and its purely from Foundry-zksync.